### PR TITLE
[3.1.0] Inline warnings

### PR DESCRIPTION
**Pull request by worlddlckgh**
- Helius DAS limitation

### CODE REVIEW

File: src/components/SuggestionTags/hooks/useSuggestionTags.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+          FREEZE_AUTHORITY_IGNORE_LIST.includes(fromTokenInfo.address) === false && // Ignore bluechip like, USDC, USDT
+          (tokenExt1.freezeAuthority === SystemProgram.programId.toString()) === false // Ignore system program
+        ) {
+          freeze.push(fromTokenInfo); // Only mark non-strict token, so USDC, USDT, don't get marked
+        }
+
+        if (tokenExt1?.permanentDelegate) {
+          permanent.push(fromTokenInfo);
+        }
+      })();
```
Comment (nghanyi) [inline src/components/SuggestionTags/hooks/useSuggestionTags.tsx:102]: probably is my copy pasted code. do we need to use IIFE here?

Comment (worlddlckgh) [reply src/components/SuggestionTags/hooks/useSuggestionTags.tsx:102]: done, thanks

File: src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+  const { pillTitle, title, description } = useMemo(() => {
+    if (isHighPriceImpact) {
+      return {
+        pillTitle: `${new Decimal(priceImpactPct).mul(100).toNumber().toFixed(2)}% Price Impac`,
+        title: shouldPromptDca ? `Looks like your trade size is too large.` : `Looks like you are getting a bad rate.`,
+        description: shouldPromptDca ? (
+          <Link
+            shallow
+            href={`/dca/${fromTokenInfo.address}-${toTokenInfo.address}?${new URLSearchParams({
```
Comment (wkitsin) [inline src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx:94]: this one is wrong, should be https://jup.ag, right now it just reloads t he widget

File: src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+          >
+            <JupButton size="sm" type="button" className="!w-[120px]">
+              <div className="flex gap-x-2 items-center justify-center">
+                <span>{`Go to DCA`}</span>
+                <RightArrowIcon />
+              </div>
+            </JupButton>
+          </Link>
+        ) : (
+          `Please refresh quote or trade with caution.`
```
Comment (gabrielkhh) [inline src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx:109]: Need to add the translations macro here? Seems like the usage in the div below never wrap translations either.

File: src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        ),
+      };
+    }
+
+    return {
+      pillTitle: `${priceDifferencePct.toFixed(2)}% Price Difference`,
+      title: `Looks like you are getting a bad rate.`,
+      description: `Please refresh quote or trade with caution.`,
```
Comment (gabrielkhh) [inline src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx:117]: This too

File: src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    () => isDanger || swapRouteInfo?.isHighImpact,
+    [isDanger, swapRouteInfo?.isHighImpact],
+  );
+
+  const isMobile = useMobile();
+
+  const { pillTitle, title, description } = useMemo(() => {
+    if (isHighPriceImpact) {
+      return {
+        pillTitle: `${new Decimal(priceImpactPct).mul(100).toNumber().toFixed(2)}% Price Impac`,
```
Comment (wkitsin) [inline src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx:89]: typo `Price Impac`

File: src/components/SuggestionTags/Tags/Token2022Info.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  if (props.isLoading) {
+    return (
+      <div className="flex justify-center my-5">
+        <Spinner />
+      </div>
+    );
+  }
+
+  if (!asset) {
+    return <div className="flex justify-center my-5 text-xs">Could not retrieve Token2022 information.</div>;
```
Comment (gabrielkhh) [inline src/components/SuggestionTags/Tags/Token2022Info.tsx:33]: Translations I guess

File: src/components/SuggestionTags/Tags/Token2022Info.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  return (
+    <div className="mt-3 mb-5">
+      <p className="text-center text-xs text-v2-lily">
+        This token utilizes the Token2022 program or Token Extension, which offer a superset of the features provided by
```
Comment (gabrielkhh) [inline src/components/SuggestionTags/Tags/Token2022Info.tsx:39]: Here too

File: src/components/SuggestionTags/Tags/UnknownTokenSuggestion.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+export const UnknownTokenSuggestion = ({ tokenInfo }: { tokenInfo: TokenInfo }) => {
+  return (
+    <PopoverTooltip
+      placement="top"
+      drawShades
+      buttonContentClassName="!cursor-help"
+      content={
+        <div className="text-xs">
+          {tokenInfo.symbol} is not on the strict list, make sure the mint address is correct before trading
```
Comment (gabrielkhh) [inline src/components/SuggestionTags/Tags/UnknownTokenSuggestion.tsx:14]: Here too

**Comment by wkitsin** [comment]
tested with these tokens 

1. unknown token - 342Ddj8upBeX7yyRa4ruhkpVhsiwfQyNoohymKNiH1Ge
2. token2022 with no TT - t3DohmswhKk94PPbPYwA6ZKACyY3y5kbcqeQerAJjmV
3. token 2022 with TT - BERN
4. token 2022 with TT but 0% - PYUSD
5. token with freeze - 5zyzuiqEXh4TPEjwgQapkrbctTYUeQdGsXBTwnD6M4mG

other than the comments, LGTM

**Pull request merged**